io.c

```diff
@@ -3203,67 +3203,90 @@ io_getpartial(int argc, VALUE *argv, VALUE io, int no_exception, int nonblock)
 
 /*
  *  call-seq:
- *     ios.readpartial(maxlen)              -> string
- *     ios.readpartial(maxlen, outbuf)      -> outbuf
+ *    readpartial(maxlen)             -> string
+ *    readpartial(maxlen, out_string) -> out_string
  *
- *  Reads at most <i>maxlen</i> bytes from the I/O stream.
- *  It blocks only if <em>ios</em> has no data immediately available.
- *  It doesn't block if some data available.
+ *  Reads up to +maxlen+ bytes from the stream.
  *
- *  If the optional _outbuf_ argument is present,
- *  it must reference a String, which will receive the data.
- *  The _outbuf_ will contain only the received data after the method call
- *  even if it is not empty at the beginning.
+ *  With the single non-negative integer argument +maxlen+ given,
+ *  returns a new string with +maxlen+ bytes from the stream, if available,
+ *  or all available bytes otherwise:
```

Comment:
The behavior with two arguments should be exactly the same as with one argument, except that in the two argument case, instead of creating a new string, the content of the two argument is replaced.  Can we make that more clear to the reader? Similar for `read`.

Reply:
Modified (for both methods).